src/main/kotlin/dev/stalla/util/AnyExtensions.kt

```diff
@@ -1,5 +1,39 @@
 package dev.stalla.util
 
+import kotlin.contracts.contract
+
+/** Check if all argument elements are not null. */
+@InternalApi
+internal fun allNotNull(a: Any?): Boolean {
+    contract {
+        returns(true) implies (a != null)
+        returns(false) implies (a == null)
+    }
+    return a != null
+}
+
+/** Check if all argument elements are not null. */
+@InternalApi
+internal fun allNotNull(a: Any?, b: Any?): Boolean {
+    contract {
+        returns(true) implies (a != null && b != null)
+    }
+    return a != null && b != null
+}
+
+/** Check if all argument elements are not null. */
+@InternalApi
+internal fun allNotNull(a: Any?, b: Any?, c: Any?): Boolean {
+    contract {
+        returns(true) implies (
+            a != null &&
+                b != null &&
+                c != null
+            )
+    }
+    return a != null && b != null && c != null
+}
+
 /** Check if all argument elements are not null. */
 @InternalApi
 internal fun allNotNull(vararg elements: Any?): Boolean = elements.all { p -> p != null }
```

Comment:
That a very good idea! But I guess the first argument shouldn't have a default value though. 

It would be so great if contracts would allow something like `returns(true) implies (*elements != null)` in our `vararg` methods...

Reply:
Actually there is a [KEEP issue](https://github.com/Kotlin/KEEP/issues/139#issuecomment-662703696) where somebody suggested exactly that. Every 👍 helps I suppose.